src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            machineList.addAll(bulldozerProducer.get());
+        }
+
+        if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            machineList.addAll(excavatorProducer.get());
+        }
+
+        if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            machineList.addAll(truckProducer.get());
+        }
+        return machineList;
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        Collections.fill(machines, value);
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        machines.forEach(Machine::doWork);
+
+    }
 }
```

Comment:
Don't begin class or method implementation with empty lines (lines 11-13). It would be better to remove these to comply with the checklist.